x-pack/test/security_solution_api_integration/test_suites/detections_response/basic_essentials_license/rule_management/coverage_overview.ts

```diff
@@ -0,0 +1,585 @@
+          );
+
+          const body = await getCoverageOverview(supertest, {
+            source: [CoverageOverviewRuleSource.Prebuilt],
+          });
+
+          expect(body).to.eql({
+            coverage: {
+              T001: [expectedRule.id],
+              TA001: [expectedRule.id],
+              'T001.001': [expectedRule.id],
+            },
+            unmapped_rule_ids: [],
+            rules_data: {
+              [expectedRule.id]: {
+                activity: 'disabled',
+                name: 'Query with a rule id',
+              },
+            },
+          });
+        });
+
+        it('returns all rules if both custom and prebuilt filters are specified in the request', async () => {
+          await createPrebuiltRuleAssetSavedObjects(es, [
+            createRuleAssetSavedObject({
+              rule_id: 'prebuilt-rule-1',
+              threat: generateThreatArray(1),
+            }),
+          ]);
+          const {
+            results: { created },
+          } = await installPrebuiltRules(es, supertest);
+          const expectedPrebuiltRule = created[0];
+
+          const expectedCustomRule = await createRule(
+            supertest,
+            log,
+            getCustomQueryRuleParams({ rule_id: 'rule-1', threat: generateThreatArray(2) })
+          );
+
+          const body = await getCoverageOverview(supertest, {
+            source: [CoverageOverviewRuleSource.Prebuilt, CoverageOverviewRuleSource.Custom],
+          });
+
+          expect(body).to.eql({
+            coverage: {
+              T001: [expectedPrebuiltRule.id],
+              TA001: [expectedPrebuiltRule.id],
+              'T001.001': [expectedPrebuiltRule.id],
+              T002: [expectedCustomRule.id],
+              TA002: [expectedCustomRule.id],
+              'T002.002': [expectedCustomRule.id],
+            },
+            unmapped_rule_ids: [],
+            rules_data: {
+              [expectedPrebuiltRule.id]: {
+                activity: 'disabled',
+                name: 'Query with a rule id',
+              },
+              [expectedCustomRule.id]: {
+                activity: 'disabled',
+                name: 'Custom query rule',
+              },
+            },
+          });
+        });
+      });
+    });
+
+    describe('error cases', async () => {
+      it('throws error when request body is not valid', async () => {
+        const { body } = await supertest
+          .post(RULE_MANAGEMENT_COVERAGE_OVERVIEW_URL)
+          .set('kbn-xsrf', 'true')
+          .set('elastic-api-version', '1')
+          .set('x-elastic-internal-origin', 'foo')
+          .send({ filter: { source: ['give me all the rules'] } })
+          .expect(400);
+
+        expect(body).to.eql({
+          statusCode: 400,
+          error: 'Bad Request',
+          message:
+            '[request body]: Invalid value "give me all the rules" supplied to "filter,source"',
+        });
```

Comment:
It's good to have one integration test for that, but let's also add unit tests for `CoverageOverviewRequestBody` that would cover the schema well.

Reply:
Added